src/Session.php

```diff
@@ -1829,6 +1829,8 @@ public static function startImpersonating($user_id)
         $auth->user = $user;
         Session::init($auth);
 
+        $_SESSION['glpiname'] = sprintf(__('%s (impersonated by %s)'), $_SESSION['glpiname'], $impersonator);
```

Comment:
We have to make sure name in session is properly reset when stopping impersonating; but this seems a good idea to add this information.
Also, translation already exists :)